packages/gatsby-theme-newrelic/src/components/GlobalHeader.js

```diff
@@ -27,7 +30,20 @@ const styles = {
 };
 
 const GlobalHeader = ({ editUrl, className, search }) => {
-  const [isOverlayOpen, setIsOverlayOpen] = useState(false);
+  const [isOverlayOpen, setIsOverlayOpen] = useState(null);
+
+  const [isQueryChanged, setIsQueryChanged] = useState(false);
+
+  useEffect(() => {
+    let searchQuery = qs.parse(window.location.search);
+    if (isOverlayOpen === null && searchQuery.overlay) setIsOverlayOpen(true);
+    if (isOverlayOpen === null && !searchQuery.overlay) setIsOverlayOpen(false);
+
+    if (isOverlayOpen && window.location.search.length && !searchQuery.overlay)
+      setQueryStringWithoutPageReload(window.location.search + '&overlay=true');
+    if (isOverlayOpen && !window.location.search.length)
+      setQueryStringWithoutPageReload('?overlay=true');
+  }, [isOverlayOpen, isQueryChanged]);
```

Comment:
It doesn't look like `isQueryChanged` is used anywhere inside this effect. Is that intentional? Could you talk about what you're trying to accomplish? 

My best guess is that between this hook and the `window.history.pushState` below, you're trying to track when the location has changed so you can determine whether to open/close the overlay. If that is the case, you might consider instead using a combination of `useEffect` with the `useLocation` hook from reach router. This will play better with React because `useLocation` is meant for location change tracking.

This might look something like this:
```js
import { useLocation } from '@reach/router'
// ...
const location = useLocation()

useEffect(() => {
  // use location instead of window.location inside of here
}, [isOverlayOpen, location])
```

I believe this might satisfy what you're trying to accomplish with the `window.history.pushState` below, which also means you'll no longer need the `isQueryChanged` variable since you'll be tracking changes to the location directly.

Reply:
So what I'm trying to do is trigger useEffect when `search-ui` changes the URL so that I can append `overlay=true` to the end of the URL. So what happens is when you search is that the search component changes the URL, and since we want to use that functionality I want to figure out when they're calling pushState (which is what the `track` function was trying to accomplish).

Reply:
Out of curiosity, does the `useLocation()` hook work here when search ui does its thang?

Reply:
It does not! it's because the search ui component is updating the window by `pushState` and `useLocation()` only gets the location on initial render. I can't put it inside `useEffect` because I can't put a hook inside of a callback. So I'm not sure I can utilize `useLocation` for this purpose.